test/e2e/e2e_workload_test.go

```diff
@@ -62,23 +63,30 @@ func TestWorkloadE2E(t *testing.T) {
 		}
 
 		// create a duplicate workload definition
-		yamlDoc := ""
 		duplicateWorkload := v0.WorkloadDefinition{
 			Definition: v0.Definition{
 				Name: &workloadDefName,
 			},
-			YAMLDocument: &yamlDoc,
+			YAMLDocument: util.StringPtr(""),
 		}
 
 		// determine if the API is serving HTTPS or HTTP
 		var authEnabled bool
 		var respCodeErr error
 		var resp *http.Response
 		// check HTTPS
-		resp, err := http.Get(fmt.Sprintf("http://localhost:%d/version", 443))
-		// if we get an error, check HTTP
-		if err != nil {
-			resp, err = http.Get(fmt.Sprintf("http://localhost:%d/version", 80))
+		if err := util.Retry(10, 1, func() error {
+			var err error
+			resp, err = http.Get(fmt.Sprintf("http://localhost:%d/version", 443))
+			// if we get an error, check HTTP
+			if err != nil {
+				resp, err = http.Get(fmt.Sprintf("http://localhost:%d/version", 80))
+				if err != nil {
+					return err
+				}
+			}
+			return nil
+		}); err != nil {
```

Comment:
Wait until API server to start in the event that it's still performing database migrations, building etc.